test/app.e2e-spec.ts

```diff
@@ -1,20 +1,307 @@
+  UPDATE_USER_DTO_WITHOUT_PASSWORD_FIELDS
+} from './test.constants';
 
 describe('AppController (e2e)', () => {
   let app: INestApplication;
+  let databaseService: DatabaseService;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [AppModule],
+      providers: [DatabaseService]
     }).compile();
 
+    databaseService = moduleFixture.get<DatabaseService>(DatabaseService);
+
     app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(
+      new ValidationPipe({
+        transform: true,
+        whitelist: true
+      })
+    );
     await app.init();
   });
 
-  afterAll(() => {
-    app.close();
+  beforeEach(async () => {
+    await databaseService.cleanAll();
+  });
+
+  describe('/users', () => {
+
+    describe('POST', () => {
+      it('should return 201 status when user is created', async () => {
+        return await request(app.getHttpServer()).post('/users').set('Content-Type', 'application/json').send(CREATE_USER_DTO_TEST_OBJ).expect(HttpStatus.CREATED).then(response => {
+          expect(response.body.createdAt.valueOf()).not.toBe(TEST_USER.createdAt.valueOf());
+          expect(response.body.email).toEqual(TEST_USER.email);
+          expect(response.body.firstName).toEqual(TEST_USER.firstName);
+          expect(response.body.id).toBeDefined();
+          expect(response.body.lastLogin).toEqual(null);
+          expect(response.body.lastName).toEqual(TEST_USER.lastName);
+          expect(response.body.loginCount).toEqual(TEST_USER.loginCount.toString());
+          expect(response.body.organization).toEqual(TEST_USER.organization);
+          expect(response.body.role).toEqual(TEST_USER.role);
+          expect(response.body.title).toEqual(TEST_USER.title);
+          expect(response.body.updatedAt.valueOf()).not.toBe(TEST_USER.updatedAt.valueOf());
+        });
+      });
+
+      it('should return 400 status if passwords dont match', async () => {
+        return await request(app.getHttpServer()).post('/users').set('Content-Type', 'application/json').send(CREATE_USER_DTO_TEST_OBJ_WITH_UNMATCHING_PASSWORDS)
+          .expect(HttpStatus.BAD_REQUEST).then(response => {
+            expect(response.body.message).toEqual('Passwords do not match');
+            expect(response.body.error).toEqual('Bad Request');
+          });
+      });
+
+      it('should return 400 status if password is not provided', async () => {
+        return await request(app.getHttpServer()).post('/users').set('Content-Type', 'application/json').send(CREATE_USER_DTO_TEST_OBJ_WITH_MISSING_PASSWORD_FIELD)
+          .expect(HttpStatus.BAD_REQUEST).then(response => {
+            expect(response.body.message[0]).toEqual('password should not be empty');
+            expect(response.body.error).toEqual('Bad Request');
+          });
+      });
+
+      it('should return 400 status if passwordConfirmation is not provided', async () => {
```

Comment:
I would like to see a test with a password that does not meet the complexity requirements.